packages/syft/src/syft/custom_worker/builder_k8s.py

```diff
@@ -0,0 +1,318 @@
+# stdlib
+from hashlib import sha256
+import os
+from pathlib import Path
+from typing import Dict
+from typing import Optional
+
+# third party
+import kr8s
+from kr8s.objects import APIObject
+from kr8s.objects import ConfigMap
+from kr8s.objects import Job
+
+# relative
+from .builder_types import BuilderBase
+from .builder_types import ImageBuildResult
+from .builder_types import ImagePushResult
+
+__all__ = ["KubernetesBuilder"]
+
+JOB_COMPLETION_TTL = 60
+
+BUILD_OUTPUT_PVC = "worker-builds"
+KUBERNETES_NAMESPACE = os.getenv("K8S_NAMESPACE", "syft")
+
+
+class InvalidImageDigest(Exception):
+    pass
+
+
+class KubernetesBuilder(BuilderBase):
+    def __init__(self):
+        self.client = kr8s.api(namespace=KUBERNETES_NAMESPACE)
+
+    def build_image(
+        self,
+        tag: str,
+        dockerfile: str = None,
+        dockerfile_path: Path = None,
+        buildargs: Optional[dict] = None,
+        **kwargs,
+    ) -> ImageBuildResult:
+        job_id = self._new_job_id(tag)
+
+        if dockerfile:
+            pass
+        elif dockerfile_path:
+            dockerfile = dockerfile_path.read_text()
+
+        # Create a ConfigMap with the Dockerfile
+        config = self._create_build_config(job_id, dockerfile)
+        config.refresh()
+
+        # Create and start the job
+        job = self._create_kaniko_build_job(
+            job_id=job_id,
+            tag=tag,
+            job_config=config,
+            build_args=buildargs,
+        )
+
+        try:
+            # wait for job to complete/fail
+            job.wait(["condition=Complete", "condition=Failed"])
+
+            # TODO: check job status, raise with logs
+
+            image_digest = self._get_image_digest(job)
+            if not image_digest:
+                raise InvalidImageDigest("Did not get any image digest from the job")
+
+            # get logs
+            logs = self._get_logs(job)
+        except Exception:
+            raise
+        finally:
+            # don't delete the job, kubernetes will gracefully do that for us
+            config.delete()
+
+        return ImageBuildResult(
+            image_hash=image_digest,
+            logs=logs,
+        )
+
+    def push_image(
+        self,
+        tag: str,
+        username: str,
+        password: str,
+        registry_url: str,
+        **kwargs,
+    ) -> ImagePushResult:
+        # Create and start the job
+        job_id = self._new_job_id(tag)
+        job = self._create_push_job(
```

Comment:
MAYBE: We could add a try catch block also for push , like we have for build_image above

Reply:
Not needed same as https://github.com/OpenMined/PySyft/pull/8417#discussion_r1469387426

Reply:
To summarize, API->Service->Handlers must return SyftSuccess/SyftError, meanwhile core modules that power the Handlers should be pure python. This makes sure the code is design to be tested in isolation and independent of any unexpected cross-deps.

Reply:
I see the try catch block is part of docker push/docker build